providers/base/units/usb/usb-c.pxu

```diff
@@ -208,3 +208,129 @@ _steps:
+     (it's the USB type C connector) of the DUT.
+_verification:
+  The host detects and mounts a mass storage device. It has read and write 
+  permissions on it.
+plugin: user-interact-verify
+command:
+  echo "Creating 16 MB image file..."
+  dd if=/dev/zero of="${PLAINBOX_SESSION_SHARE}"/checkbox-mass-storage-test.img bs=1M count=16
+  echo "Formatting image file as FAT32..."
+  mkdosfs -F 32 "${PLAINBOX_SESSION_SHARE}"/checkbox-mass-storage-test.img
+  echo "loading g_mass_storage module..."
+  modprobe g_mass_storage file="${PLAINBOX_SESSION_SHARE}"/checkbox-mass-storage-test.img stall=0
+user: root
+category_id: com.canonical.plainbox::usb
+estimated_duration: 5m
+flags: preserve-locale also-after-suspend
+imports: from com.canonical.plainbox import manifest
+requires: manifest.has_usbc_otg == 'True'
+
+id: usb-c-otg/g_mass_storage-cleanup
+_summary: Cleanup mass storage setup after mass storage device test
+plugin: shell
+after: usb-c-otg/g_mass_storage
+command:
+  echo "Removing g_mass_storage module..."
+  modprobe -r g_mass_storage
+  if [ -f "${PLAINBOX_SESSION_SHARE}/checkbox-mass-storage-test.img" ]; then
+    echo "Removing image file..."
+    rm "${PLAINBOX_SESSION_SHARE}"/checkbox-mass-storage-test.img
+  fi
+user: root
+estimated_duration: 1s
+category_id: com.canonical.plainbox::usb
+flags: also-after-suspend
+
+id: usb-c-otg/g_ether
+_summary: Check DUT can be detected as USB ethernet device
+_steps:
+  1. From an Ubuntu host, connect a cable to the USB OTG port 
+     (it's the USB type C connector) of the DUT.
+  2. Press Enter to config the IP address for the USB OTG ethernet interface on the DUT.
+     (IP will be 192.168.9.1/24)
+  3. On the host, config the IP address for the USB OTG ethernet interface. And it should be 
+     the same subnet with DUT. (eg. 192.168.9.2/24)
+  4. Try to ping between the host and DUT.
+_verification:
+  The host and DUT can ping each other.
+plugin: user-interact-verify
+command:
+   echo "Setting up the IP address for USB OTG ethernet interface..."
+   modprobe g_ether
+   if  (ip -c a | grep -q usb0) ; then 
+      echo "USB OTG ethernet interface exist. Configuring..."
+      ip addr add 192.168.9.1/24 dev usb0
+      echo "USB OTG ethernet interface IP address been set as 192.168.9.1/24"
+      ip link set dev usb0 up
+   else 
+      echo "USB OTG ethernet interface does not exist."
+      exit 1
+   fi
+user: root
+category_id: com.canonical.plainbox::usb
+estimated_duration: 5m
+flags: preserve-locale also-after-suspend
+imports: from com.canonical.plainbox import manifest
+requires: manifest.has_usbc_otg == 'True'
+
+id: usb-c-otg/g_ether-cleanup
+_summary: Cleanup USB OTG ethernet interface setup after ethernet device test
+plugin: shell
+after: usb-c-otg/g_ether
+command:
+  echo "Removing g_ether module..."
+  modprobe -r g_ether
+user: root
+estimated_duration: 1s
+category_id: com.canonical.plainbox::usb
+flags: also-after-suspend
```

Comment:
Similar to my comment above for the serial test, you will have to set things manually with the `siblings` field if you want to point to the right `after` job.